sql/core/v2.3/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcColumnVector.java

```diff
@@ -136,7 +138,9 @@ public int getInt(int rowId) {
   public long getLong(int rowId) {
     int index = getRowIndex(rowId);
     if (isTimestamp) {
-      return timestampData.time[index] * 1000 + timestampData.nanos[index] / 1000 % 1000;
```

Comment:
I took a look at ORC type spec but it doesn't mention the calendar. The physical timestamp type looks very similar to `Timestamp`, so this looks correct to me.

How about the write side?

Reply:
The write side uses toJavaTimestamp already: https://github.com/apache/spark/blob/300ec1a74cb14867c22e616e657566d510426331/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcSerializer.scala#L144-L148 . So after this PR, it will do rebasing automatically

Reply:
Yes, @cloud-fan . Apache ORC inherited Apache Hive's original design.
- https://github.com/apache/hive/blob/master/storage-api/src/java/org/apache/hadoop/hive/ql/exec/vector/TimestampColumnVector.java#L46-L50